### DISCUSSION TITLE
Using OpenGlControlBase with a non-cooperative renderer

### DISCUSSION THREAD

**Discussion by jgcodes2020** [AvaloniaUI/Avalonia#11926] [Q&A]
I'm writing an emulator frontend for [Mupen64Plus](https://github.com/mupen64plus/mupen64plus-core) (hereafter M64+). M64+ renders its graphics to OpenGL, though unlike some other libraries, it needs to decide when to swap buffers.

While my initial solution was just to use `NativeControlHost` and SDL to create a window, I need to be able to render an overlay on top of the emulator's output. For that, I need to render the OpenGL output to an actual control, as opposed to a hosted window. How do I do that?

**Comment by kekekeks** [comment]
You can use your own context and present surfaces via https://github.com/AvaloniaUI/Avalonia/issues/9925

**Reply by jgcodes2020** [reply]
Thanks for that! Is there a sample somewhere?

**Reply by jgcodes2020** [reply]
https://github.com/AvaloniaUI/Avalonia/discussions/11926#discussioncomment-6298377
@kekekeks adding on to this, how should I create the shared OpenGL context?

**Reply by kekekeks** [reply]
( await _compositor.TryGetRenderInterfaceFeature(typeof(IOpenGlTextureSharingRenderInterfaceContextFeature))?.CreateSharedContext
